### PR TITLE
refactor: move backup module and input module to coreutils_core

### PR DESCRIPTION
**Pull request by Larisho**
Unfortunately, I can't manage to get several of the coreutils_core modules to build on my machine. I'm hoping that TravisCI will be able to build it successfully.

### CODE REVIEW

**Comment by Larisho** [comment]
@GrayJack I only added files to the core. Do you have an idea of what's happening here? I'm at a loss

**Comment by GrayJack** [comment]
Looks like you didn't put the modules on lib.rs

**Comment by GrayJack** [comment]
NetBSD and Solaris worked out cause in their toml there is no mv, so they don't error out

File: coreutils_core/src/input.rs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+impl Input {
+    pub fn new() -> Self {
+        let mut line = String::new();
+        match io::stdin().lock().read_line(&mut line) {
+            Ok(_) => {},
+            Err(err) => {
+                eprintln!("rm: cannot read input: {}", err);
+                process::exit(1);
+            },
```
[the remark this answers is not in the document]
Comment (Larisho) [reply coreutils_core/src/input.rs]: You're absolutely right--Missed it in the copy/paste

File: coreutils_core/src/backup.rs [hunk 2]
```diff
@@ -0,0 +1,79 @@
+use regex::Regex;
+use std::{
+    fs,
+    io::{Error, ErrorKind},
+    path::PathBuf,
+};
+
+#[derive(Debug, Clone, PartialEq)]
```
[the remark this answers is not in the document]
Comment (Larisho) [reply coreutils_core/src/backup.rs]: Do you have a set of particular derived traits you want?

File: mv/Cargo.toml [hunk 3]
```diff
@@ -6,6 +6,7 @@ edition = "2018"
 
 [dependencies]
 clap = { version = "^2.33.0", features = ["yaml", "wrap_help"] }
+coreutils_core = {path = "../coreutils_core"}
 regex = "^1"
```
[the remark this answers is not in the document]
Comment (Larisho) [reply mv/Cargo.toml]: Why remove it? How will it find the module?

File: coreutils_core/src/backup.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (GrayJack) [reply coreutils_core/src/backup.rs]: Derive everything that std have, remove the ones that the compiler complains

File: mv/Cargo.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (GrayJack) [reply mv/Cargo.toml]: I don't saw where mv was using Regex

**Comment by Larisho** [comment]
FYI: Am adding documentation if all the checks pass

File: coreutils_core/src/input.rs [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+/// ```
+#[derive(Debug, Clone)]
+pub struct Input<'a> {
+    msg:     Option<&'a str>,
+    err_msg: Option<&'a str>,
+}
+
+impl<'a> Default for Input<'a> {
+    fn default() -> Input<'a> { Input { msg: None, err_msg: None } }
+}
```
Comment (GrayJack) [inline coreutils_core/src/input.rs]: Option always defaults to `None`, since both default to `None`, can't we derive default?

Comment (Larisho) [reply coreutils_core/src/input.rs]: Honestly, I didn't know that Default could be derived! Let me look into it and see. Expect another review request!

**Comment by Larisho** [comment]
> Just a note: You're way better at documenting than I'll ever be 👍

Thanks! I do way too much of it at work so that might be why I am so well practiced 😂

**Pull request merged**